### PR TITLE
Removed obsole part of scheduler that caused a crash

### PR DESCRIPTION
**Pull request by spirali**
Fixes: https://github.com/It4innovations/hyperqueue/issues/673

### CODE REVIEW

File: crates/tako/src/internal/tests/test_scheduler_sn.rs [hunk 1]
```diff
@@ -206,101 +206,101 @@ fn test_no_deps_distribute_with_balance() {
...
-    comm.take_worker_msgs(100, 1);
-    comm.take_worker_msgs(101, 1);
-
-    check_task_has_worker(&core, 13, 100);
-    check_task_has_worker(&core, 14, 101);
-
-    comm.emptiness_check();
-    core.sanity_check();
-}
+// #[test]
```
Comment (Kobzol) [inline crates/tako/src/internal/tests/test_scheduler_sn.rs:209]: Come one, let's delete it :) We can find it in git history if needed.

File: tests/pyapi/test_dependencies.py [hunk 2]
```diff
@@ -38,3 +40,20 @@ def test_dep_failed(hq_env: HqEnv):
     assert table.get_row_value("1") == "CANCELED"
     assert table.get_row_value("2") == "CANCELED"
     assert table.get_row_value("3") == "FINISHED"
+
+
+def test_kill_worker_with_deps(hq_env: HqEnv):
```
Comment (Kobzol) [inline tests/pyapi/test_dependencies.py:45]: This test succeeds on `main`.

**Review by Kobzol — APPROVED** [review]
Great, even looks like a small perf. win on 10k-sleep:
```
Grouped by environment:
hq tag=current,worker_count=1,zero_worker=False
	sleep duration=0,task_count=10: 0.0621 s (min=0.0595, max=0.0648)
	sleep duration=0,task_count=100: 0.1659 s (min=0.1477, max=0.1840)
	sleep duration=0,task_count=1000: 0.5154 s (min=0.4969, max=0.5340)
	sleep duration=0,task_count=10000: 5.0719 s (min=5.0381, max=5.1057)
(mean): 1.4538 s

hq tag=main,worker_count=1,zero_worker=False
	sleep duration=0,task_count=10: 0.0508 s (min=0.0484, max=0.0532)
	sleep duration=0,task_count=100: 0.0982 s (min=0.0961, max=0.1003)
	sleep duration=0,task_count=1000: 0.5111 s (min=0.5029, max=0.5193)
	sleep duration=0,task_count=10000: 5.1543 s (min=5.0669, max=5.2417)
(mean): 1.4536 s
```

Left one nit.

File: tests/pyapi/test_dependencies.py (hunk 2 above)
Comment (spirali) [reply tests/pyapi/test_dependencies.py:45]: I know, I am not not able to trigger it on main. But as I have written this test I just put it in repo

File: crates/tako/src/internal/tests/test_scheduler_sn.rs (hunk 1 above)
Comment (spirali) [reply crates/tako/src/internal/tests/test_scheduler_sn.rs:209]: Ooookeeeyyy

File: tests/pyapi/test_dependencies.py (hunk 2 above)
Comment (Kobzol) [reply tests/pyapi/test_dependencies.py:45]: Ok :)

**Pull request merged**